### PR TITLE
Trampoline transform

### PR DESCRIPTION
**Pull request by EPashkin**
Part of https://github.com/gtk-rs/gir/issues/264
Applying trampoline transformation in https://github.com/gtk-rs/gir/pull/293

WIP

Borrow:
`TreeView`: added signal `move-cursor`
...
TreePath:
`CellRendererCombo::changed`: +borrow
`CellArea`: +borrow, `add-editable` also has `Rectangle`

### CODE REVIEW

**Comment by EPashkin** [comment]
@GuillaumeGomez, @gkoz  take look at regen samples.

File: src/auto/cell_area.rs [hunk 1]
```diff
@@ -112,9 +121,17 @@ pub trait CellAreaExt {
...
+
+    fn connect_apply_attributes<F: Fn(&Self, &TreeModel, &TreeIter, bool, bool) + 'static>(&self, f: F) -> u64;
+
+    fn connect_focus_changed<F: Fn(&Self, &CellRenderer, TreePath) + 'static>(&self, f: F) -> u64;
+
+    fn connect_remove_editable<F: Fn(&Self, &CellRenderer, &CellEditable) + 'static>(&self, f: F) -> u64;
 }
 
-impl<O: IsA<CellArea>> CellAreaExt for O {
+impl<O: IsA<CellArea> + IsA<Object>> CellAreaExt for O {
```
Comment (GuillaumeGomez) [inline src/auto/cell_area.rs]: If `IsA<CellArea>` is "true", then shouldn't `IsA<Object>` be "true" as well?

File: src/auto/cell_renderer_accel.rs [hunk 2]
```diff
@@ -1,9 +1,12 @@
-// This file was generated by gir (4d68d19) from gir-files (11e0e6d)
+// This file was generated by gir (bc9cb02) from gir-files (???)
```
Comment (GuillaumeGomez) [inline src/auto/cell_renderer_accel.rs]: Why has it become '???'?

**Comment by GuillaumeGomez** [comment]
So now we can generate trampolines as well? 😂

File: src/auto/cell_area.rs (hunk 1 above)
Comment (EPashkin) [inline src/auto/cell_area.rs]: Currently - not, so we adding it to all traits with signal.
You seen this in #394

File: src/auto/cell_renderer_accel.rs (hunk 2 above)
Comment (EPashkin) [inline src/auto/cell_renderer_accel.rs]: Its rust's git bug with submodule `objects/info/alternates` in msys.
I make it normal when ready to merge.

**Comment by GuillaumeGomez** [comment]
Well, I don't have much to say on this PR. Good job!

**Comment by EPashkin** [comment]
In `signal.rs` only last `Editable` and `SpinButton` with too specific code

**Comment by GuillaumeGomez** [comment]
It's been around for a week and like it's ok for me. Still no news from @gkoz.  If it's ok for you @EPashkin, I'll merge this (waiting for confirmation).

**Comment by EPashkin** [comment]
No, this is not ready for merge.
Stalled by https://github.com/gtk-rs/gir/pull/293 (its better not be merged without @gkoz review), and 293 based on https://github.com/gtk-rs/gir/pull/292 that IMHO you can merge.

**Comment by GuillaumeGomez** [comment]
Ok, I merge the mergeable one and wait for the other.

**Comment by EPashkin** [comment]
Rebased, but still not ready

**Comment by EPashkin** [comment]
Ready for merge

**Comment by GuillaumeGomez** [comment]
Waiting for appveyor and then I merge.

Thanks a lot for your work! :)

**Comment by EPashkin** [comment]
Restarted appveyor (error due network failure)

**Pull request merged**